app/factories/shopify/product_image_factory.rb

```diff
@@ -0,0 +1,38 @@
+module Shopify
+  class ProductImageFactory
+    def initialize(spree_variant)
+      @spree_variant = spree_variant
+    end
+
+    def perform
+      shopify_image = ShopifyAPI::Image.new
+
+      shopify_image.variant_ids = [spree_variant.pos_variant_id]
+      shopify_image.attachment = encoded_image(variant_image)
+
+      shopify_image
+    end
+
+    private
+
+    def encoded_image(image)
+      # When the image is hosted locally, we must get the `path` of the image,
+      # else if it's hosted online we must get the `url` of the image.
+      # I haven't found a better way of doing that yet.
+      begin
```

Comment:
Clever! You might consider using `image.attachment.exists?` and `image.attachment.copy_to_local_file`, which seems to be an  interface method you can rely on in any of Paperclip's storage backends:

- [S3][1]
- [Fog][2]
- [Filesystem][3]

[1]: https://github.com/thoughtbot/paperclip/blob/master/lib/paperclip/storage/s3.rb#L408
[2]: https://github.com/thoughtbot/paperclip/blob/master/lib/paperclip/storage/fog.rb#L169
[3]: https://github.com/thoughtbot/paperclip/blob/master/lib/paperclip/storage/filesystem.rb#L84-L86